### PR TITLE
Standardize dogfood and small fixes

### PR DESCRIPTION
**Pull request by ydubreuil**
Replaces #523. I had to push my branch to jenkinsci repository in order to get it built by PR builder.

This is a follow-up to JENKINS-37141.

Depends on a fixed Jenkins LTS version instead of latest to ensure reproducible builds.
Don't bypass installation wizard by default as it's here to enforce security.

With this, Docker image built from master branch should be usable for dogfooding, which would make https://github.com/cloudbees/blueocean-docker-image obsolete

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by ydubreuil** [comment]
I'd like to merge this quickly so people starting to work on new things could base their PR on top of this.

File: pom.xml [hunk 1]
```diff
@@ -18,7 +18,7 @@
   <properties>
     <java.level>7</java.level>
     <jackson.version>2.2.3</jackson.version>
-    <jenkins.version>2.7.1</jenkins.version>
+    <jenkins.version>2.7.1</jenkins.version> <!-- Should be kept in sync with Dockerfile FROM statement-->
```
Comment (bwalding) [inline pom.xml]: Should this be 2.7.4?

**Comment by michaelneale** [comment]
I think this is ok :bee: - seems like it won't harm anything existing

**Comment by tfennelly** [comment]
🐝 LGTM

**Comment by michaelneale** [comment]
(failure is npm being a poo poo head, ignore).

**Pull request merged**